include/SFML/Window/Event.hpp

```diff
@@ -181,6 +182,18 @@ class Event
     };
 
     ////////////////////////////////////////////////////////////
+    /// \brief File dropping parameters (FilesDropped)
+    ///
+    ////////////////////////////////////////////////////////////
+    struct FileDropEvent
+    {
+        unsigned int count;  ///< Number of dropped files
+        const String* files; ///< Pointer to the first file, access with event.droppedFiles.files[i] (in range [0 .. event.droppedFiles.count - 1])
```

Comment:
All event data is stored in a `union`, so using `std::vector<string>` is not possible. Since C++11, it would theoretically be allowed using placement new and manual destruction invocations. Alternatively, there's also `std::variant`/`boost::variant`.